backend/src/packages/users/user.repository.ts

```diff
@@ -61,25 +67,35 @@ class UserRepository implements Repository {
       return null;
     }
 
+    const subscriptionEndDate = user.details?.subscription?.endDate
+      ? new Date(user.details.subscription.endDate)
+      : null;
+
     return UserEntity.initialize({
       id: user.id,
       email: user.email,
       createdAt: new Date(user.createdAt),
       updatedAt: new Date(user.updatedAt),
       fullName: user.details?.fullName ?? '',
       isSurveyCompleted: user.details?.isSurveyCompleted ?? false,
+      subscriptionId: user.details?.subscriptionId ?? null,
+      subscriptionEndDate,
     });
   }
 
   public async findAll(): Promise<UserWithPasswordEntity[]> {
```

Comment:
do we need findAll in users?

Reply:
as I see, this method was created by you in the project starter. I can remove it, if we don't need it

Reply:
it is not used right now

Reply:
remove it please